crates/typst-library/src/foundations/dict.rs

```diff
@@ -266,6 +268,64 @@ impl Dict {
             .map(|(k, v)| Value::Array(array![k.clone(), v.clone()]))
             .collect()
     }
+
+    /// Produces a new dictionary with only the pairs from the original one for
+    /// which the given function returns `{true}`.
+    ///
+    /// ```example
```

Comment:
Let's add a title here as well. The current policy is to always have titles for all examples as soon as there is more than one example.